src/main/java/com/openlattice/datastore/search/controllers/SearchController.java

```diff
@@ -432,94 +415,85 @@ public SearchResult executeFQNPropertyTypeSearch( @RequestBody FQNSearchTerm sea
     public Map<UUID, List<NeighborEntityDetails>> executeEntityNeighborSearchBulk(
             @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
             @RequestBody Set<UUID> entityKeyIds ) {
-        return executeFilteredEntityNeighborSearch( entitySetId, new EntityNeighborsFilter( entityKeyIds ) );
+        return executeFilteredEntityNeighborSearch( new EntityNeighborsFilterBulk( Map.of( entitySetId, entityKeyIds ) ) );
     }
 
     @RequestMapping(
-            path = { ENTITY_SET_ID_PATH + NEIGHBORS + ADVANCED },
+            path = { NEIGHBORS + ADVANCED },
             method = RequestMethod.POST,
             produces = { MediaType.APPLICATION_JSON_VALUE } )
     @Override
     @Timed
     public Map<UUID, List<NeighborEntityDetails>> executeFilteredEntityNeighborSearch(
-            @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
-            @RequestBody EntityNeighborsFilter filter ) {
+            @RequestBody EntityNeighborsFilterBulk filter
+    ) {
         Set<Principal> principals = Principals.getCurrentPrincipals();
+        var entitySetIds = filter.getEntityKeyIds().keySet();
 
         Map<UUID, List<NeighborEntityDetails>> result = Maps.newHashMap();
-        if ( authorizations.checkIfHasPermissions( new AclKey( entitySetId ), principals,
-                EnumSet.of( Permission.READ ) ) ) {
-
-            EntitySet es = entitySetManager.getEntitySet( entitySetId );
-
-            checkState( es != null, "Could not find entity set with id: " + entitySetId.toString() );
-
-            final var entitySets = ( es.isLinking() ) ? es.getLinkedEntitySets() : Set.of( entitySetId );
-            final var authorizedEntitySets = entitySets.stream()
-                    .filter( linkedEntitySetId ->
-                            authorizations.checkIfHasPermissions( new AclKey( linkedEntitySetId ),
-                                    Principals.getCurrentPrincipals(),
-                                    EnumSet.of( Permission.READ ) ) )
-                    .collect( Collectors.toSet() );
-            if ( authorizedEntitySets.size() != entitySets.size() ) {
-                logger.warn( "Read authorization failed some of the normal entity sets of linking entity set or it " +
-                        "is empty." );
-            } else {
-                result = searchService
-                        .executeEntityNeighborSearch( ImmutableSet.of( entitySetId ), filter, principals );
-            }
+        if ( getEntitySetsForRead( entitySetIds, principals ) != null ) {
+            result = searchGraphService.executeEntityNeighborSearch( filter, principals );
         }
 
-
         /* audit */
 
-        ListMultimap<UUID, UUID> neighborsByEntitySet = ArrayListMultimap.create();
+        Map<UUID, List<UUID>> neighborsByEntitySet = Maps.newHashMap();
```

Comment:
as above, `result.values().size * neighborsList.size` to at least not have to resize for a bit

Reply:
👍